src/main/java/seedu/mark/model/annotation/PhantomParagraph.java

```diff
@@ -41,26 +60,60 @@ public ParagraphContent getParagraphContent() {
     }
 
     /**
-     * Vacuously true; phantom paragraph has empty string as content so highlight does not matter.
-     * TODO: rethink bout this
+     * Returns true. The only key part of annotation is a note. (Highlight is ignored)
      */
     @Override
-    public boolean hasHighlight() {
+    public boolean hasAnnotation() {
+        assert note != null : "Phantom has no annotations...";
```

Comment:
Same comment about assertions...
```suggestion
        if (note == null) {
            throw new ParagraphException("Phantom has no annotations...");
        }
```

Reply:
Actually I'm not really sure whether we should use assertions or exceptions; my original intention of using assertions was to make sure nothing goes wrong when we create and use Phantom Paragraphs; by right such a scenario will not happen if we implemented it correctly

for this case, I'm trying to assert that `PhantomParagraph` can never exist without an annotation after a visible state change reflected on gui.

Reply:
Ah I see. You're right, assertions make more sense here. 

Do you think we'll need to document these assumptions/requirements in the Javadocs too?